.github/actions/sdk-codegen/publish.sh

```diff
@@ -0,0 +1,111 @@
+#
+# ============================================================
+
+git config credential.helper '!f() { sleep 1; echo "username=${GITHUB_USER}"; echo "password=${GITHUB_TOKEN}"; }; f'
+git config --global user.name "Algorand Generation Bot"
+git config --global user.email "$GITHUB_EMAIL"
+
+# ========================================================
+# > EXPORT DATA FROM SPEC FILES
+# ========================================================
+
+pushd "$GO_ALGORAND_DIR" > /dev/null
+export ALGOD_SPEC_HASH=`git log -n 1 --pretty=format:%h -- daemon/algod/api/algod.oas2.json`
+export ALGOD_SPEC_COMMIT_MSG=`git log -n 1 --pretty=format:%s -- daemon/algod/api/algod.oas2.json`
+popd > /dev/null
+pushd "$INDEXER_DIR" > /dev/null
+export INDEXER_SPEC_HASH=`git log -n 1 --pretty=format:%h -- api/indexer.oas2.json`
+export INDEXER_SPEC_COMMIT_MSG=`git log -n 1 --pretty=format:%s -- api/indexer.oas2.json`
+popd > /dev/null
+
+# ==============================
+# > CREATE BRANCH
+# ==============================
+
+# Generate an identifier for this PR (first four chars of algod/indexer hashes, concatenated)
+PR_ID="${ALGOD_SPEC_HASH:0:4}${INDEXER_SPEC_HASH:0:4}"
+PR_BRANCH="generate/$PR_ID"
+
+# Don't continue if a remote branch has already been created
+REMOTE_PR_BRANCH_EXISTS=`git ls-remote origin $PR_BRANCH | wc -l`
+if [ $REMOTE_PR_BRANCH_EXISTS -eq 1 ]; then
+  echo "Generated branch already exists, stopping"
+  exit 0
+fi
+
+# Delete the branch if it exists locally
+PR_BRANCH_EXISTS=`git branch -v | grep $PR_BRANCH | wc -l`
+if [ $PR_BRANCH_EXISTS -eq 1 ]; then
+  git -c core.hooksPath=/dev/null checkout $BRANCH # disable git hooks https://stackoverflow.com/a/61485071
+  git branch -D $PR_BRANCH > /dev/null
+fi
+
+# Create a new branch
+git -c core.hooksPath=/dev/null checkout -b $PR_BRANCH # disable git hooks https://stackoverflow.com/a/61485071
+
+# Exit without error if there are no changes to commit
+CHANGED=`git diff --name-only HEAD --`
+if [ -z "$CHANGED" ]; then
+  echo "No new changes, exiting"
+  exit 0
+fi
+
+# Create commit and store hash
+git add .
+git commit --no-verify -m "Regenerate code from specification file"
+
+# Push
+git -c core.hooksPath=/dev/null push --set-upstream origin $PR_BRANCH # disable git hooks https://stackoverflow.com/a/61485071
+
+# ==============================
+# > CREATE A PR
+# ==============================
+
+# Design PR content
+ALGOD_REPO_URL="https://github.com/algorand/go-algorand"
+INDEXER_REPO_URL="https://github.com/algorand/indexer"
+
+PR_BODY="This PR was automatically created using [Algorand's code generator](https://github.com/algorand/generator), in response to the following commits:
+
+### Algod
+
+ - [$ALGOD_SPEC_HASH]($ALGOD_REPO_URL/commit/$ALGOD_SPEC_HASH) – $ALGOD_SPEC_COMMIT_MSG
+
+### Indexer
+
+ - [$INDEXER_SPEC_HASH]($INDEXER_REPO_URL/commit/$INDEXER_SPEC_HASH) - $INDEXER_SPEC_COMMIT_MSG
+
+
+> **Disclaimer:** I'm just a bot. Feel free to make changes to this pull request as needed."
```

Comment:
Nice touch with this PR body and links to the spec

Reply:
Thanks but I can't really take credit as I copied this over from https://github.com/algorand/generator/blob/master/automation/publish.sh#L109 😸